### PR TITLE
Lps 106347 delete experience

### PR DESCRIPTION
**Pull request by andresfulla**

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js [hunk 1]
```diff
@@ -60,3 +62,72 @@ export function recoverModalExperienceState() {
 
 	return null;
 }
+
+function distinct(element, index, array) {
+	return array.indexOf(element) === index;
+}
+
+function onlyFragmentType(element) {
```
Comment (dgarciasarai) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js]: `isFragmentType`? 🤔

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js [hunk 2]
```diff
@@ -60,3 +62,72 @@ export function recoverModalExperienceState() {
...
+		.map(({config}) => config.fragmentEntryLinkId);
+
+	const layoutDatasWithoutExcluded = layoutDataList
+		.filter(({segmentsExperienceId: id}) => {
+			return id !== experienceIdToExclude;
+		})
+		.map(({layoutData: layout}) => {
+			if (selectedExperienceId) {
+				/**
+				 * Updated layoutData
```
Comment (dgarciasarai) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js]: This is a TODO?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js [hunk 3]
```diff
@@ -60,3 +62,72 @@ export function recoverModalExperienceState() {
...
+		)
+		.filter(onlyFragmentType)
+		.map(({config}) => config.fragmentEntryLinkId)
+		.filter(distinct);
+
+	const notInOtherLayoutsFilter = notInArray(
+		fragmentEntryLinksIdsInNotExcluded
+	);
+
+	const uniqueFragmentEntryLinkIdsToLayout = fragmentEntryLinkIdsInLayout.filter(
```
Comment (dgarciasarai) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js]: `uniqueFragmentEntryLinksIds` missing an 's' in word Link(s) to be consistent

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js [hunk 4]
```diff
@@ -35,6 +35,7 @@ import ExperienceToolbarSection from '../../../../../../src/main/resources/META-
 
 import '@testing-library/jest-dom/extend-expect';
 
+const MOCK_DELETE_URL = 'delete-experience-test-url';
```
Comment (dgarciasarai) [inline modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js]: Sort?

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js [hunk 5]
```diff
@@ -478,4 +480,137 @@ describe('ExperienceToolbarSection', () => {
...
+				items: {
+					'00001': {
+						config: {
+							fragmentEntryLinkId: 1000
+						},
+						type: 'fragment'
+					},
+					'00004': {
+						config: {
+							fragmentEntryLinkId: 4000 // latest version of layoutData is not in layoutDataList
```
Comment (dgarciasarai) [inline modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js]: Another TODO?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js (hunk 1 above)
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js]: ♥

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js (hunk 5 above)
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js]: no, this is a fragment that is not in the layoutDataList of that experience. A reference

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js (hunk 4 above)
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js]: not in the sorting business

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js (hunk 1 above)
Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js]: +1

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js (hunk 3 above)
Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/utils.js]: +1

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js (hunk 5 above)
Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js]: 00000000000000000000000000000000000000000000000004

**Comment by p2kmgcl** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/83330

**Pull request closed without merging**